wrapanapi/base/template.py

```diff
@@ -0,0 +1,35 @@
+"""
+Methods/classes pertaining to performing actions on a template
+
+"""
+from abc import ABCMeta, abstractmethod, abstractproperty
+
+from .entity import BaseEntity
+
+
+class BaseTemplate(BaseEntity):
+    __metaclass__ = ABCMeta
+
+    @property
+    def exists(self):
+        """Checks if a template of this name exists
+
+        Returns:
+            True if it exists
+            False if not
+            False if system's list_templates() method is not implemented.
```

Comment:
Is a ```False``` here the right action? I'm wondering if we should be letting the Error hit. This could yield us to try to deploy a template if we don't see it, or something like that.

Reply:
Sure, I'll raise NotImplementedError instead. And also fix the loop logic here because it's not right ;)